plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -186,7 +214,7 @@ func (c *X509Cert) serverName(u *url.URL) (string, error) {
 	return u.Hostname(), nil
 }
 
-func (c *X509Cert) getCert(u *url.URL, timeout time.Duration) ([]*x509.Certificate, error) {
+func (c *X509Cert) getCert(u *url.URL, timeout time.Duration) (*X509CertificatesOcsp, error) {
```

Comment:
Can we please just return the OCSP reponse as another return item instead of constructing/deconstructing that new struct? If possible, even parse it here into the expected struct...